plugins/azure-devops-backend/src/utils/azure-devops-utils.ts

```diff
@@ -205,6 +205,32 @@ export function convertPolicy(
   };
 }
 
+export async function replaceReadme(
+  readme: string,
+  getFileContent: (
+    path: string,
+    encoding?: BufferEncoding,
+  ) => Promise<{
+    url: string;
+    content: string;
+  }>,
+) {
+  const filesPath = extractAssets(readme);
+  return await filesPath.reduce(
+    async (promise: Promise<string>, filePath: string) =>
+      promise.then(async content => {
+        const { label, path, ext } = extractPartsFromAsset(filePath);
+        const mime = getMimeByExtension(ext);
+        const file = await getFileContent(path + ext, 'base64');
+        return content.replace(
+          filePath,
+          `[${label}](data:${mime};base64,${file.content})`,
```

Comment:
Hm. So you're going through this whole ordeal to avoid CORS/CSP errors, is that it? I'm a bit wary of it tbh, these will start taking some time to load and they might get overly large sometimes etc. Could you explain the motivation here just so I understand the need so to speak?

Reply:
I work with @alissonfabiano and I can answer this one as it was my suggestion.

ADO doesn't give us a public URL to raw git files like GitHub does with `https://raw.githubusercontent.com/`. The initial plan was to proxy all images through backstage's backend, this would open a back door into all (or most) of our git repos by using the proxy.